core/src/main/scala/sttp/openai/json/SttpOpenAIApi.scala

```diff
@@ -0,0 +1,69 @@
+package sttp.openai.json
+
+import sttp.client4.json._
+import sttp.client4.upicklejson.SttpUpickleApi
+import sttp.client4._
+import sttp.model.StatusCode._
+import sttp.model.ResponseMetadata
+import sttp.openai.OpenAIExceptions.OpenAIException
+import sttp.openai.OpenAIExceptions.OpenAIException._
+
+/** An sttp upickle api extension that deserializes JSON with snake_case keys into case classes with fields corresponding to keys in
+  * camelCase and maps errors to OpenAIException subclasses.
+  */
+object SttpOpenAIApi extends SttpUpickleApi {
```

Comment:
I think the previous name was better, we're defining here upickle extensions to work with open ai? while we are not defining an open ai API?